src/Select.js

```diff
@@ -1,141 +1,112 @@
-import React, { Component, Children, cloneElement } from 'react';
+import React, {
+  Children,
+  cloneElement,
+  useRef,
+  useEffect,
+  useState
+} from 'react';
 import PropTypes from 'prop-types';
 import cx from 'classnames';
 import idgen from './idgen';
 import constants from './constants';
 
-class Select extends Component {
-  constructor(props) {
-    super(props);
-
-    this.id = props.id || idgen();
-    this.handleChange = this.handleChange.bind(this);
-    this.state = { value: props.value };
-  }
-
-  handleChange(event) {
-    const { onChange } = this.props;
-    let value = event.target.value;
-
-    if (this.props.multiple) {
-      value = this.instance.getSelectedValues();
-    }
+const Select = ({
+  id,
+  s,
+  m,
+  l,
+  xl,
+  noLayout,
+  browserDefault,
+  icon,
+  label,
+  className,
+  success,
+  error,
+  validate,
+  children,
+  multiple,
+  options,
+  value,
+  onChange,
+  ...props
+}) => {
+  const [selectedValue, setSelectedValue] = useState(value);
```

Comment:
do we want to update the value if it's passed via props again?

Reply:
I think we should.
Otherwise, we might have a situation where the state is out of sync.
Think of `controlled component` pattern, the component will receive the `selected` value as `prop`.
What do you think?

Reply:
ping @alextrastero :smile: